File: 003/C/003.c
#include <stdio.h>

void main(){
    long a=600851475143, b, c;
    for (b=2; b<=a; b++){
        if (a%b==0) {
        a = a/b;
        c = b;
        b--;
        if(a==1)
            break;
        }
    }
    printf("Largest one: %d\n", c);    
}

File: 003/go/003.go
// run in http://go-vim.appspot.com/p/wN3Cy4lDy9
package main

import "fmt"

func main() {
	n := 600851475143
	i := 2
	for i*i < n {
		for n%i == 0 {
			n = n / i
		}
		i++
	}
	fmt.Printf("The largest prime factor of the number 600851475143 is %d", n)
}
